langchain/llms/alpaca.py

```diff
@@ -0,0 +1,159 @@
+# Credit to https://github.com/nsarrazin/serge
+# - this is heavily copied from the API there and not very well yet but it might work.
+import asyncio
+import subprocess
+import threading
+from datetime import datetime
+from typing import Any, AsyncIterable, List, Mapping, Optional
+from uuid import UUID, uuid4
+
+from pydantic import BaseModel, Field
+
+from langchain.llms.base import BaseLLM, Generation, LLMResult
+
+
+class ChatParameters(BaseModel):
```

Comment:
There's a lot of shared `ChatParameters` between this, LlammaCPP, and GPT4All. Probably OK to pref duplicate code over sharing given the nature of the files though

Reply:
@vowelparrot I didn't realise there was a LlamaCpp class because last time I looked it wasn't there. This code makes no sense, an alias or a thin wrapper would make sense. I just wasted a bunch of time by not looking again, after doing a bunch of nice tidying [here](https://github.com/hwchase17/langchain/compare/harrison/alpaca...lukestanley:langchain:harrison/alpaca?expand=1), I was about to rename it to Llamma when I noticed that LlamaCpp exists already! D'oh! I better see what stuff it makes sense to port to the LlamaClass.